client/landing/stepper/declarative-flow/internals/steps-repository/completing-purchase/index.tsx

```diff
@@ -39,60 +40,84 @@ const CompletingPurchase: Step = function CompletingPurchase( { navigation, flow
 		}
 
 		// convert ArrayBuffer to Blob
-		const blob = new Blob( [ buffer ], { type: type } );
-
-		return blob;
+		return new Blob( [ buffer ], { type: type } );
 	};
 
-	const completeLinkInBioFlow = () => {
-		if ( site ) {
-			setPendingAction( async () => {
-				setProgress( 0 );
-				setProgressTitle( __( 'Completing Purchase' ) );
-				const state = getState();
-				await saveSiteSettings( site.ID, {
-					blogname: state.siteTitle,
-					blogdescription: state.siteDescription,
-				} );
-
-				recordTracksEvent( 'calypso_signup_site_options_submit', {
-					has_site_title: !! state.siteTitle,
-					has_tagline: !! state.siteDescription,
-				} );
-				setProgress( 0.5 );
-				setProgressTitle( __( 'Creating your Link in Bio' ) );
-
-				await wait( 1000 );
-
-				setProgress( 0.8 );
-				setProgressTitle( __( 'Preparing Next Steps' ) );
-
-				if ( state.siteLogo ) {
-					try {
-						await setSiteLogo(
-							new File( [ base64ImageToBlob( state.siteLogo ) ], 'site-logo.png' )
-						);
-					} catch ( _error ) {
-						// communicate the error to the user
-					}
-				}
-				setProgress( 1 );
-			} );
+	const postSiteSettings = useCallback( async () => {
+		if ( ! site ) {
+			return;
+		}
+
+		await saveSiteSettings( site.ID, {
+			blogname: state.siteTitle,
+			blogdescription: state.siteDescription,
+		} );
+
+		recordTracksEvent( 'calypso_signup_site_options_submit', {
+			has_site_title: !! state.siteTitle,
+			has_tagline: !! state.siteDescription,
+		} );
+	}, [ site, saveSiteSettings, state ] );
+
+	const postSiteLogo = useCallback( async () => {
+		if ( ! state.siteLogo ) {
```

Comment:
I am not sure if it is a mandatory field, If so we need to find a way to communicate to the user, as we don't have 'goBack' buttons.. I think we could validate it on the setup step.

Perhaps this is out of scope for this PR..